services/src/contexts/mod.rs

```diff
@@ -343,3 +380,85 @@ where
         }
     }
 }
+
+#[cfg(test)]
+
+mod tests {
+    use super::*;
+    use std::path::PathBuf;
+
+    use geoengine_datatypes::{test_data, util::test::TestDefault};
+
+    use crate::{
+        contexts::{Context, InMemoryContext, SimpleSession},
+        util::config::set_config,
+    };
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+    async fn read_model_test() {
+        // TODO: find a better solution for this
+        // necessary to avoid getting a wrong config in the other related tests
+        let ten_millis = std::time::Duration::from_millis(100);
+        std::thread::sleep(ten_millis);
+
+        set_config(
+            "machinelearning.model_defs_path",
+            test_data!("pro/ml").to_str().unwrap(),
+        )
+        .unwrap();
+
+        let ctx = InMemoryContext::test_default();
+        let exe_ctx = ctx.execution_context(SimpleSession::default()).unwrap();
+
+        let model_path = PathBuf::from("xgboost/s2_10m_de_marburg/model.json");
+        let mut model = exe_ctx.read_ml_model(model_path).await.unwrap();
+
+        let actual: String = model.drain(0..277).collect();
+        let expected = "{\"learner\":{\"attributes\":{},\"feature_names\":[],\"feature_types\":[],\"gradient_booster\":{\"model\":{\"gbtree_model_param\":{\"num_parallel_tree\":\"1\",\"num_trees\":\"16\",\"size_leaf_vector\":\"0\"},\"tree_info\":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],\"trees\":[{\"base_weights\":[5.192308E-1,9.722222E-1";
+
+        set_config(
+            "machinelearning.model_defs_path",
+            "./test_data/pro/ml".to_string(),
```

Comment:
better save the original value before overwriting it. Otherwise there might be a conflict if it is changed in the Settings-default, but not here

Reply:
Added a backup variable [here](https://github.com/geo-engine/geoengine/pull/639/commits/15511c1bf666395c74b1bd17f8927fcf0cf0ad1a#diff-b9fca937b91582b0887a2fea16060f5f906844f8a96587673f0f8cdf3882ef87R415-R416)